### DISCUSSION TITLE
Failed to test proxy

### DISCUSSION THREAD

**Discussion by 1ReallyLongName** [FlareSolverr/FlareSolverr#1114] [Q&A] [answered]
Hello, I created the FlareSolverr container using docker-compose and I am able to access the website manually; however, it will not be found within Prowlarr. This is my first GitHub discussion, so please let me know if I broke any best practices, or if I can provide you anything else.

When I manually go to the address I see this: {"msg": "FlareSolverr is ready!", "version": "3.3.16", "userAgent": "Mozilla/5.0 (X11; Linux x86_64) AppleWebKit/537.36 (KHTML, like Gecko) Chrome/120.0.0.0 Safari/537.36"} 

docker-compose: https://privatebin.net/?4e48eb83c697af6e#CPAg5vrUf8GZtKCPeu5JrjkkvHqX8URSJU2hqz1NEa6E

QNAP NAS TVS-682: Firmware version: QTS 5.1.5.2679
Prowlarr version: 1.13.3.4273
FlareSolver version: 3.3.16

**Comment by ilike2burnthing** [comment]
What are you using for FlareSolverr's API URL in Prowlarr?

Are you using a proxy or VPN?

**Reply by 1ReallyLongName** [reply]
I hope I'm understanding the question correctly, when attempting to map the proxy the host address I'm using is http://myiphere:8191/

I am using a VPN; however, disabling it did not change the result.

**Reply by ilike2burnthing** [reply]
As both are running on Docker, try using the Docker container IP address and port (e.g. `http://172.17.0.2:8191`), or the container name and port (e.g. `http://flaresolverr:8191`).

**Reply by 1ReallyLongName** [reply]
Tested with VPN on and then off. Still returning the same error. 

![Screenshot_20240308_195754_Chrome](https://github.com/FlareSolverr/FlareSolverr/assets/134996609/4c31f22d-2042-4f3e-826e-89e449315c5f)

![Screenshot_20240308_195830_Chrome](https://github.com/FlareSolverr/FlareSolverr/assets/134996609/680f4e23-93e0-4c2b-9c95-8312908864da)

**Reply by 1ReallyLongName** [reply]
The flaresolverr one was almost immediate compared to the 127 one. I don't know if that means anything.

**Reply by ilike2burnthing** [reply]
Check Prowlarr's logs, see if it provides more insight into the actual issue. If not, enable trace logging and try again.

**Reply by 1ReallyLongName** [reply]
Here's what appears to be recurring within the logs:

2024-03-09 03:21:17.8|Error|FlareSolverr|Proxy Health Check failed

[v1.13.3.4273] System.Net.WebException: Http request timed out
   at NzbDrone.Common.Http.Dispatchers.ManagedHttpDispatcher.GetResponseAsync(HttpRequest request, CookieContainer cookies) in ./Prowlarr.Common/Http/Dispatchers/ManagedHttpDispatcher.cs:line 162
   at NzbDrone.Common.Http.HttpClient.ExecuteRequestAsync(HttpRequest request, CookieContainer cookieContainer) in ./Prowlarr.Common/Http/HttpClient.cs:line 171
   at NzbDrone.Common.Http.HttpClient.ExecuteAsync(HttpRequest request) in ./Prowlarr.Common/Http/HttpClient.cs:line 70
   at NzbDrone.Common.Http.HttpClient.Execute(HttpRequest request) in ./Prowlarr.Common/Http/HttpClient.cs:line 137
   at NzbDrone.Core.IndexerProxies.FlareSolverr.FlareSolverr.Test() in ./Prowlarr.Core/IndexerProxies/FlareSolverr/FlareSolverr.cs:line 188

**Reply by ilike2burnthing** [reply]
Provide the trace log of this.

**Reply by 1ReallyLongName** [reply]
Attached. I ran all urls we had discussed. 

[prowlarr.trace.txt](https://github.com/FlareSolverr/FlareSolverr/files/14545041/prowlarr.trace.txt)

**Reply by ilike2burnthing** [reply]
From within the Prowlarr container, run `curl -v http://172.17.0.2:8191` and provide the output.

**Reply by 1ReallyLongName** [reply]
docker exec prowlarr curl -v http://172.17.0.2:8191
  % Total    % Received % Xferd  Average Speed   Time    Time     Time  Current
                                 Dload  Upload   Total   Spent    Left  Speed
  0     0    0     0    0     0      0      0 --:--:-- --:--:-- --:--:--     0*   Trying 172.17.0.2:8191...
  0     0    0     0    0     0      0      0 --:--:--  0:02:10 --:--:--     0* connect to 172.17.0.2 port 8191 from 172.29.0.8 port 37614 failed: Operation timed out
* Failed to connect to 172.17.0.2 port 8191 after 130305 ms: Couldn't connect to server
  0     0    0     0    0     0      0      0 --:--:--  0:02:10 --:--:--     0
* Closing connection
curl: (28) Failed to connect to 172.17.0.2 port 8191 after 130305 ms: Couldn't connect to server

**Reply by ilike2burnthing** [reply]
Whoops, my bad, that should have been `curl -v http://172.29.4.2:8191`

**Reply by 1ReallyLongName** [reply]
I was wondering. Sadly the exact same result.

**Reply by ilike2burnthing** [reply]
Are you currently running Prowlarr and FlareSolverr as Docker swarm? If so, just set them up as two separate containers and see if that makes a difference.

**Reply by 1ReallyLongName** [reply]
I'm not even too sure what a docker swarm is so I'm going to bet I do not have it set up as a swarm 😅

Looking in my container station they are separate containers though, and I have them separated in my docker compose.

**Reply by ilike2burnthing** [reply, marked as answer]
Looking a little closer at your compose, the hostname for the FlareSolverr container is actually `7cf070e5f81c` rather than `flaresolverr`, so you can try `http://7cf070e5f81c:8191` (in Prowlarr, and if that doesn't work then with curl).

---

I've typed that and then read further. You have Prowlarr on the `media_default` network and FlareSolverr on the `trash-guides_default` network, so I'm going to guess that's why they can't communicate.

**Reply by 1ReallyLongName** [reply]
I see that now and believe you're absolutely right. I'm super new to docker-compose so I'm very confused because I never designated each of them different networks. Also, I'm assuming it's because I made edits to my original docker compose, but now it won't let me close any of my previous ones containers through ```docker compose down```. Just this new one.

**Reply by 1ReallyLongName** [reply]
You were absolutely right! It seems way too obvious now. For some reason it created two separate ones when I installed flaresolverr separately. So I had to remove all of them through container station and then rerun docker compose. Thank you for all your help and sorry for the goose chase!
